README


This archive contains the distribution MooseX-Types,
version 0.45:

  Organise your Moose types in libraries

This software is copyright (c) 2007 by Robert "phaylon" Sedlacek.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for MooseX-Types

0.45      2014-09-02 01:17:50Z
        - increase the required versions of some prerequisites

0.44      2014-03-09 04:16:20Z
        - full respect for namespaces: declared types are now properly
          installed into the proper package, so they survive
          namespace::autoclean; however, types imported from the declaration
          class are still properly cleaned by namespace::autoclean (as
          implemented in trial versions 0.40-0.43)

0.43      2014-01-17 01:25:55Z (TRIAL RELEASE)
        - line numbers in shipped code are now almost the same (within 3) as
          the repository source, for easier debugging
        - skipped new namespacing tests that fail with older Moose versions,
          due to an inability to install blessed subs

0.42      2014-01-04 23:31:58Z (TRIAL RELEASE)
        - declared (not imported) types are now properly installed into the
          proper package, so they survive namespace::autoclean (fixed this
          time)

0.41      2013-12-21 17:21:41Z
        - reversed changes in 0.40, until compatibility issues with older
          Moose versions are figured out

0.40      2013-12-20 06:38:15Z
        - types are now properly installed into the proper package, so they
          survive namespace::autoclean

0.39      2013-11-30 18:31:32Z
        - more strict warnings checking in tests
        - re-release to fix compile test

0.38      2013-09-17 00:44:16Z
        - removed use of deprecated enum syntax

0.37      2013-09-08 21:58:26Z
        - removed use of deprecated Class::MOP::load_class
        - repository has moved to the GitHub Moose organization

0.36    2013-06-22 20:51:54Z
        - fixed failing tests on 5.8.5 due to incorrect == overload
        (RT#77100, Graham Knop)

0.35    2012-05-19
        - The new delegation code could fail when a type had no
          parent. Attempting to subtype Any was one way to trigger the
          bug. (Dave Rolsky)

        - The new delegation code broke calling ->can or ->isa on
          MooseX::Types::TypeDecorator as a class method. (Dave Rolsky)

0.34    2012-05-18
        - Special case new to avoid unexpected exceptions on unloaded classes

0.33    2012-05-18
        - Switch delegation order to prioritise type over class except for new

0.32    2012-05-18
        - Support delegation of methods to the class for class types
        - Factor out _try_delegate method

0.31    2011-12-22
        - Add support for qw( :all ) on MooseX::Types::Combine libraries. (kentnl)

0.30    2011-09-17
        - Fix a bug in the tests that caused failures with Moose HEAD. (doy)

0.29    2011-08-20
        - Require namespace::clean 0.19. Tests fail with 0.18. Reported by
          Ruslan Zakirov. RT #67923.

0.28    2011-08-02
        - Fixed git and bug tracking info in META.{yml,json}. Reported by
          Petter Rabbitson. (Dave Rolsky)

0.27    2011-06-06
        - A better fix for type constraint changes in Moose 2.0100. (doy)

0.26    2011-06-05
        - Small changes for forward compatibility with Moose 2.0100, when it
          is released. (Dave Rolsky)
        - Some small doc cleanup and improvements. (Dave Rolsky)

0.25    2010-11-30
        - The test suite now uses Test::Fatal instead of Test::Exception
          (Karen Etheridge).
        - Modernized tests to use Test::Requires and done_testing. (Dave
          Rolsky)
        - MooseX::Types::Combine is a little more efficient, and only tries to
          load the libraries it's combining when that list is set. It also
          checks that each lib in the list actually provides types. (Dave
          Rolsky).
        - The MooseX::Types docs now contain a pointer to
          MooseX::Types::Combine. (Dave Rolsky)

0.24    2010-09-11
        - Avoid more warnings from Moose 1.09 in tests (Dave Rolsky).

0.23    2010-07-19
        - Avoid warnings from Moose 1.09 in tests (Dave Rolsky).

0.22    2010-06-01
        - Update docs to mention MRO::Compat rather than Class::C3.
        - Added '0+' overloading to solve the type constraint equality
          regression introduced when Moose 1.05 tightened the rules
          determing type constraint equality.  Added tests for the
          regression.

0.21    2009-12-23
        - Attempting to import a non-existent type from a combined type
          library gave a very confusing error message from the
          MooseX::Types::Combined internals. Now it gives you a much more
          useful message. (Dave Rolsky)
        - bump up Moose dep (RT#53016)

0.20    2009-09-10
        - fix "empty export" warning (doy).

0.19    2009-08-23
        - un-fix coercions on parameterized types, since it broke
          MooseX::Method::Signatures, as well as any other code that relied on
          new type constraint objects being generated each time a type was
          parameterized (hdp).

0.18    2009-08-22
        - fix coercions on parameterized types (Hans Dieter Pearcey).

0.17    2009-08-18
        - Documentation typo fix (Dave Rolsky).
        - Stop blowing up in has_available_type_export if the introspected code
          symbol exists but is not a type export (Florian Ragwitz).
        - Reorganize author/copyright sections at the request of Debian
          packagers (Rafael Kitover).

0.16    2009-06-29
        - Ship with Module::Install 0.91

0.15    2009-06-27
        - Change all uses of croak in TypeDecorator to use the Moose
          throw_exception method.
        - More test cases for the union mixed string type issue, and
          better handling of this in the | overload in TypeDecorator

0.14    2009-06-26
        - Fix union with string type

0.13    2009-06-23
        - Add Test::Moose to build_requires for Fedora

0.12    2009-06-14
        - fix mixing of typelibs with regular Moose classes

0.11    2009-05-23
        - warning on string types and unregistered class/role types
        - better error for missing comma

Build.PL
# This Build.PL for MooseX-Types was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.005
if (eval 'use Module::Build::Tiny 0.037; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for MooseX-Types was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.007.
    use strict;
    use warnings;

    use 5.008;
    # use Module::Build::Tiny 0.037;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.037"
      },
      "dist_abstract" => "Organise your Moose types in libraries",
      "dist_author" => [
        "Robert \"phaylon\" Sedlacek <rs\@474.at>"
      ],
      "dist_name" => "MooseX-Types",
      "dist_version" => "0.45",
      "license" => "perl",
      "module_name" => "MooseX::Types",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Carp" => 0,
        "Carp::Clan" => "6.00",
        "Exporter" => 0,
        "Module::Runtime" => 0,
        "Moose" => "1.06",
        "Moose::Exporter" => 0,
        "Moose::Meta::TypeConstraint::Union" => 0,
        "Moose::Util::TypeConstraints" => 0,
        "Scalar::Util" => "1.19",
        "Sub::Exporter" => 0,
        "Sub::Name" => 0,
        "base" => 0,
        "namespace::autoclean" => "0.16",
        "overload" => 0,
        "perl" => "5.008",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "Moose::Role" => 0,
        "Test::Fatal" => 0,
        "Test::More" => "0.88",
        "Test::Requires" => 0,
        "if" => 0,
        "lib" => 0
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "Moose::Role" => 0,
      "Test::Fatal" => 0,
      "Test::More" => "0.88",
      "Test::Requires" => 0,
      "if" => 0,
      "lib" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}
